src/worker/browser-env/runner/test-runner/index.ts

```diff
@@ -59,15 +67,114 @@ export class TestRunner extends NodejsEnvTestRunner {
+                const ExecutionThreadCls = wrapExecutionThread(this._socket, throwIfAborted);
+
+                return super.runRunnables(ExecutionThreadCls) as Promise<Error | undefined>;
+            });
+        } catch (err) {
+            error = err as Error;
+        }
+
+        while (error instanceof AbortError) {
+            error = undefined;
+
+            try {
+                await this._waitBroInitOnReconnect();
+
+                error = await this._runWithAbort<Error | undefined>(throwIfAborted => {
+                    const ExecutionThreadCls = wrapExecutionThread(this._socket, throwIfAborted);
+
+                    return new Promise((resolve, reject) => {
+                        (super.runRunnables(ExecutionThreadCls) as Promise<Error | undefined>)
+                            .then(resolve)
+                            .catch(reject);
+                    });
+                });
+            } catch (err) {
+                error = err as Error;
+            }
+        }
+
+        const results = await super.finishRun(error);
         this._socket.emit(WorkerEventNames.finalize);
 
+        if (error) {
+            throw error;
+        }
+
         return results;
     }
 
+    private _runWithAbort<T>(cb: (throwIfAborted: () => void) => Promise<T>): Promise<T> {
+        const controller = new AbortController();
+        const { signal } = controller;
+
+        let isAborted = signal.aborted;
+
+        signal.addEventListener("abort", () => {
+            isAborted = true;
+        });
+
+        const throwIfAborted = (): void => {
+            if (isAborted) {
+                throw new AbortError();
+            }
+        };
+
+        this._socket.once(BrowserEventNames.reconnect, () => {
+            this._broInitResOnReconnect = undefined;
+
+            this._socket.once(BrowserEventNames.initialize, errors => {
+                this._broInitResOnReconnect = errors;
+            });
+
+            if (this._browser.state.onReplMode) {
+                RuntimeConfig.getInstance().replServer.close();
+            }
+
+            this._isReconnected = true;
+            controller.abort();
+        });
+
+        return cb(throwIfAborted);
+    }
+
+    private _waitBroInitOnReconnect(): Promise<void> {
+        let intervalId: NodeJS.Timeout | null = null;
+
+        return new P((resolve, reject) => {
+            intervalId = setInterval(() => {
+                if (_.isUndefined(this._broInitResOnReconnect)) {
+                    return;
+                }
+
+                intervalId && clearInterval(intervalId);
+
+                _.isEmpty(this._broInitResOnReconnect) ? resolve() : reject(this._broInitResOnReconnect[0]);
+            }, BRO_INIT_INTERVAL_TIMEOUT_ON_RECONNECT).unref();
+        })
+            .timeout(
+                BRO_INIT_TIMEOUT_ON_RECONNECT,
+                `Browser didn't connect to the Vite server after reconnect in ${BRO_INIT_TIMEOUT_ON_RECONNECT}ms`,
```

Comment:
Why could that happen? Can we tell the user "what is wrong" information and/or "how to fix it"? 
That message itself only could be usefull to Testplane developers
P.S.: if you don't know any scenarios where it can happen, its ok to leave it be as it is.

Reply:
It can happen if something stuck in the browser and I didn't get request from it. I think this is a rare case and in most cases users will not see it at all. But unfortunately I can't tell the user exactly what caused the error.